### PR TITLE
feat(example): add examples of effects not based on the Actions stream

### LINKED ISSUE #1830: Example: Add examples of effects not based on the Actions stream
Add more Observable based effects to show that Effects aren't restricted to listening to the `Actions` stream in order to return new actions. Some ideas:

- A session timeout effect that is trigger when the user is idle for some amount of time. If the timeout expires, it dispatches an action to log out the user that is handled in `AuthEffects`
- Listening to router events to update the page title on successful navigation. This would include adding page titles to the route configuration.

Open to other ideas about how to show this.

### PR DESCRIPTION
**Pull request by adrianfaciu**
As suggested, added two examples of effects not based on Actions stream:
- listen for router navigation events and update the page title
- log out the user after a specified period of inactivity

Closes #1830

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Closes #1830

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by timdeschryver** [comment]
Another idea that pops to my mind would be to use google analytics. 
We could NgRx-ify Stephen's post https://fluin.io/blog/google-analytics-with-tag-manager-and-angular

**Comment by timdeschryver** [comment]
An example that still uses the actions but are not dispatching an action back to the store:
Showing a [snackbar](https://material.angular.io/components/snack-bar/overview) when a book has been added/removed.

File: projects/example-app/src/app/core/actions/user.actions.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import { createAction } from '@ngrx/store';
+
+export const idle = createAction('[User] Idle');
```
Comment (brandonroberts) [inline projects/example-app/src/app/core/actions/user.actions.ts]:
```suggestion
export const idleTimeout = createAction('[User] Idle Timeout');
```

File: projects/example-app/src/app/core/effects/router.effects.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import { Title } from '@angular/platform-browser';
+import { tap, filter, map, mergeMap } from 'rxjs/operators';
+
+@Injectable()
+export class RouterEffects {
+  updateTitle$ = createEffect(
+    () =>
+      this.router.events.pipe(
+        filter(event => event instanceof NavigationEnd),
+        map(() => this.activatedRoute),
```
Comment (brandonroberts) [inline projects/example-app/src/app/core/effects/router.effects.ts]: Combine the 2 maps into 1

File: projects/example-app/src/app/core/effects/router.effects.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        map(route => {
+          while (route.firstChild) route = route.firstChild;
+          return route;
+        }),
+        mergeMap(route => route.data),
+        map(data => data['title']),
```
Comment (brandonroberts) [inline projects/example-app/src/app/core/effects/router.effects.ts]:
```suggestion
        map(data => `Book Collection - ${data['title'])}`,
```

File: projects/example-app/src/app/core/effects/router.effects.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import { Injectable } from '@angular/core';
+import { createEffect } from '@ngrx/effects';
+import { Router, NavigationEnd, ActivatedRoute } from '@angular/router';
```
Comment (brandonroberts) [inline projects/example-app/src/app/core/effects/router.effects.ts]: Move all `@angular` imports to the top of the file, following by `rxjs` and `@ngrx`

File: projects/example-app/src/app/core/effects/user.effects.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+@Injectable()
+export class UserEffects {
+  clicks$ = fromEvent(document, 'click');
+  keys$ = fromEvent(document, 'keydown');
+  mouse$ = fromEvent(document, 'mousemove');
+
+  idle$ = createEffect(() =>
+    merge(this.clicks$, this.keys$, this.mouse$).pipe(
+      switchMapTo(timer(60 * 1000)),
```
Comment (brandonroberts) [inline projects/example-app/src/app/core/effects/user.effects.ts]:
```suggestion
      switchMapTo(timer(5 * 60 * 1000)), // 5 minute inactivity timeout
```

**Comment by adrianfaciu** [comment]
Updated with all the requested changes and added tests for user and router effects.
If we need more examples, I can add some effects for the snackbar and look into google analytics.

File: projects/example-app/src/app/core/effects/router.effects.ts [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+import { tap, filter, map, mergeMap } from 'rxjs/operators';
+
+import { createEffect } from '@ngrx/effects';
+
+@Injectable()
+export class RouterEffects {
+  updateTitle$ = createEffect(
+    () =>
+      this.router.events.pipe(
```
Comment (alex-okrushko) [inline projects/example-app/src/app/core/effects/router.effects.ts]: We have store-router package installed in the example app. Wouldn't that be better to listen to those changes?
@brandonroberts

Comment (adrianfaciu) [reply projects/example-app/src/app/core/effects/router.effects.ts]: I thought about that when I was writing the code :) The idea was to show how one would create effects not based on the Actions stream, so I went for this instead.

Comment (brandonroberts) [reply projects/example-app/src/app/core/effects/router.effects.ts]: I think these are good for what we asked for

**Pull request merged**